XML.java

```diff
@@ -332,6 +332,37 @@ public static JSONObject toJSONObject(String string) throws JSONException {
         }
         return jo;
     }
+    
+    /**
+     * Convert a well-formed (but not necessarily valid) XML string into a
+     * JSONObject. Some information may be lost in this transformation because
+     * JSON is a data format and XML is a document format. XML uses elements,
+     * attributes, and content text, while JSON uses unordered collections of
+     * name/value pairs and arrays of values. JSON does not does not like to
+     * distinguish between elements and attributes. Sequences of similar
+     * elements are represented as JSONArrays. Content text may be placed in a
+     * "content" member. Comments, prologs, DTDs, and <code>&lt;[ [ ]]></code>
+     * are ignored.
+     * 
+     * @param string 
+     *            The source string.
+     * @param boolean 
+     *            flag saying whether to maintain order or not.
+     * @return A JSONObject containing the structured data from the XML string.
+     * @throws JSONException
+     */
+    public static JSONObject toJSONObject(String string, boolean preserveOrder) throws JSONException {
+        if (!preserveOrder) {
+        	return toJSONObject(string);
+        } else {
+        	JSONObject jo = new OrderedJSONObject();
```

Comment:
As I see we have new JSONObject() creation inside XML#parse https://github.com/stleary/JSON-java/blob/28e09dc4938e65bbf022302972559ac0179b5ebc/XML.java#L327 method and your changes doesn't cover this case.